### PR TITLE
chore: add cSpell for spell checking

### PR DESCRIPTION
**Pull request by nschonni**
Add cSpell and fix minor typos

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "addrs",
+    "anotherhashishere",
+    "arcanis",
+    "Builtins",
+    "complexm",
+    "endregion",
+    "entrypoints",
+    "filesystems",
+    "hashishere",
+    "Hierachic",
```
Comment (nschonni) [inline .cspell.json:14]: This is actually a typo in ModulesInHierachicDirectoriesPlugin, but will be a breaking change to fix

File: lib/ResolverFactory.js [hunk 2]
```diff
@@ -280,7 +280,7 @@ exports.createResolver = function (options) {
 
 	resolver.ensureHook("resolve");
 	resolver.ensureHook("internalResolve");
-	resolver.ensureHook("newInteralResolve");
+	resolver.ensureHook("newInternalResolve");
```
Comment (alexander-akait) [inline lib/ResolverFactory.js:283]: Potential breaking change, better to add deprecation

Comment (nschonni) [reply lib/ResolverFactory.js:283]: How would a depreciation work for this one?
PS: did a quick search and it doesn't look like it's referenced anywhere else https://github.com/search?q=org%3Awebpack+Interal&type=code

Comment (alexander-akait) [reply lib/ResolverFactory.js:283]: This package used not only in webpack, many tools use it for resolving

**Comment by sokra** [comment]
Thanks

**Pull request merged**